release/changes.md

```diff
@@ -1 +1 @@
-- [NEW] TBD
\ No newline at end of file
+[NEW] - Drop support for fetching build scan data from Gradle Enterprise versions older than 2023.3.2
```

Comment:
This is the same wording as when we released `2.0`: https://github.com/gradle/gradle-enterprise-build-validation-scripts/releases/tag/v2.0